### ISSUE TITLE
Potential error for the binary_cross_entropy_with_logits

### ISSUE THREAD

**Issue by henghuiz-zz** [carpedm20/NTM-tensorflow#17]
In the script 'ops.py', the binary_cross_entropy_with_logits is defined by using the following equations

> For brevity, let `x = logits`, `z = targets`.  
> The logistic loss is loss(x, z) = - sum_i (x[i] * log(z[i]) + (1 - x[i]) * log(1 - z[i])).

But I think the meaning of x and z is wrong. x should be target and z should be logits. Here is the reference 
http://deeplearning.net/software/theano/library/tensor/nnet/nnet.html#theano.tensor.nnet.nnet.binary_crossentropy

**Comment by lsy643** [comment]
@zhhtc200 
I agree with you. More than that, since probability is used in cross entropy, I think x should  go after a sigmoid layer also.

**Comment by carpedm20** [comment]
Thanks. At the very first time when I wrote codes with TensorFlow, I used wrong `binary_cross_entropy_with_logits` for most of the repos and now I need to fix this all.

**Comment by henghuiz-zz** [comment]
@lsy643 
I don't think it is a good way to put a sigmoid function for x. Since x is the output of a LSTM cell, the value is bounded in [-1, 1] already. So adding a sigmoid function is not appropriate, through I tried the tf.nn.sigmoid_cross_entropy_with_logits in my code and worked very well. (I just thought sigmoid_cross_entropy_with_logits is the categorical crossentropy before. And I was wrong.)

I think a better way may be to let the input be in range [-1, 1] and use tf.squared_difference. Let's try and see.

**Comment by carpedm20** [comment]
@zhhtc200 @lsy643 I think this problem is fixed with #18 and checked its working.
